File: Tests/SkeletonCppParserTests/Fixtures/sample.cpp
class Animal {
public:
    std::string name;
    int age;

    Animal(std::string name, int age) : name(name), age(age) {}

    virtual void print() {
        std::cout << name << std::endl;
    }

    std::string greet(std::string greeting) {
        return greeting + " " + name;
    }
};

class Dog : public Animal {
public:
    std::string breed;

    Dog(std::string name, int age, std::string breed)
        : Animal(name, age), breed(breed) {}

    bool fetch(std::string item) {
        return true;
    }
};

struct Point {
    double x;
    double y;

    double distance(Point other) {
        return 0.0;
    }
};

enum Color {
    Red,
    Green,
    Blue,
};

union Value {
    int intVal;
    float floatVal;
    char charVal;
};

class Outer {
public:
    int x;

    class Inner {
    public:
        int y;
    };
};

File: Tests/SkeletonJavaParserTests/Fixtures/Sample.java
package com.example;

public interface Printable {
    void print();
    String format(String template);
}

public class Animal implements Printable {
    private String name;
    private int age;

    public Animal(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public void print() {
        System.out.println(name);
    }

    public String format(String template) {
        return String.format(template, name);
    }

    public String getName() {
        return name;
    }
}

public class Dog extends Animal {
    private String breed;

    public Dog(String name, int age, String breed) {
        super(name, age);
        this.breed = breed;
    }

    public boolean fetch(String item) {
        return true;
    }
}

public enum Color {
    RED, GREEN, BLUE
}
